docs: update rendering error page (#22523)
nuxt/nuxt: docs/1.getting-started/8.error-handling.md

## Rendering an Error Page

When Nuxt encounters a fatal error (any unhandled error on the server, or an error created with `fatal: true` on the client) it will either render a JSON response (if requested with `Accept: application/json` header) or trigger a full-screen error page.

This error may occur during the server lifecycle when:

- processing your Nuxt plugins
- rendering your Vue app into HTML
- a server API route throws an error

An error can also occur on the client side when:

- processing your Nuxt plugins 

> - before mounting the application (`app:beforemount` hook)

Reviewer: ```suggestion
- before mounting the application (`app:beforeMount` hook)
```

- mounting your app if the error was not handled with `onErrorCaptured` or `vue:error` hook
- the Vue app is initialized and mounted in browser (`app:mounted`).

> The lifecycle hooks are listed in 
> ::{link="/docs/api/advanced/hooks"}::

Reviewer: ```suggestion
The lifecycle hooks are listed [here](/docs/api/advanced/hooks).
```

> You can customize this error page by adding `~/error.vue` in the source directory of your application, alongside `app.vue`. Although it is called an 'error page' it's not a route and shouldn't be placed in your `~/pages` directory. This page has a single prop - `error` which contains an error for you to handle.

Reviewer: ```suggestion
You can customize this error page by adding `~/error.vue` in the source directory of your application, alongside `app.vue`. Although it is called an 'error page' it's not a route and shouldn't be placed in your `~/pages` directory. For the same reason, you shouldn't use `definePageMeta` within this page.

 The error page has a single prop - `error` which contains an error for you to handle.
```

> The `error` object provides the fields: `url`, `statusCode`, `statusMessage`, `message`, `description` and `data`. In case you have an error with custom fields they will be lost. For custom errors we highly recommend to use `onErrorCaptured` composable than can be called in a page/component setup function or `vue:error` runtime nuxt hook that should be configured in a nuxt plugin.

Reviewer: ```suggestion
The `error` object provides the fields: `url`, `statusCode`, `statusMessage`, `message`, `description` and `data`. If you have an error with custom fields they will be lost; you should assign them to `data` instead. For custom errors we highly recommend to use `onErrorCaptured` composable that can be called in a page/component setup function or `vue:error` runtime nuxt hook that can be configured in a nuxt plugin.
```

An error can also occur on the client side when:

- processing your Nuxt plugins 
- before mounting the application (`app:beforeMount` hook)
- mounting your app if the error was not handled with `onErrorCaptured` or `vue:error` hook
- the Vue app is initialized and mounted in browser (`app:mounted`).

The lifecycle hooks are listed [here](/docs/api/advanced/hooks).

You can customize this error page by adding `~/error.vue` in the source directory of your application, alongside `app.vue`. Although it is called an 'error page' it's not a route and shouldn't be placed in your `~/pages` directory. For the same reason, you shouldn't use `definePageMeta` within this page.

 The error page has a single prop - `error` which contains an error for you to handle.

The `error` object provides the fields: `url`, `statusCode`, `statusMessage`, `message`, `description` and `data`. If you have an error with custom fields they will be lost; you should assign them to `data` instead. For custom errors we highly recommend to use `onErrorCaptured` composable that can be called in a page/component setup function or `vue:error` runtime nuxt hook that can be configured in a nuxt plugin.

> ### Example
> 

Reviewer: ```suggestion
```
